analyze_foldamers/ensembles/cluster.py

```diff
@@ -353,18 +372,40 @@ def get_cluster_medoid_positions_DBSCAN(
     
     # Create distance matrices within each cluster:
     distances_k = {}
-    for k in range(n_clusters):
-        distances_k[k] = np.zeros((cluster_sizes[k],cluster_sizes[k]))
-        for i in range(cluster_sizes[k]):
-            for j in range(cluster_sizes[k]):
-                distances_k[k][i,j] = distances[cluster_indices[k][i],cluster_indices[k][j]]
     
-    # Compute medoid based on similarity scores:
-    medoid_index = []
-    for k in range(n_clusters):
-        medoid_index.append(
-            np.exp(-distances_k[k] / distances_k[k].std()).sum(axis=1).argmax()
-        )
+    if core_points_only:
+        for k in range(n_clusters):
+            distances_k[k] = np.zeros((cluster_sizes_core[k],cluster_sizes_core[k]))
+            for i in range(cluster_sizes_core[k]):
+                for j in range(cluster_sizes_core[k]):
+                    distances_k[k][i,j] = distances[cluster_indices_core[k][i],cluster_indices_core[k][j]]
+        
+        # Compute medoid based on similarity scores:
+        medoid_index = [] # Global index
+        intra_cluster_medoid_index = [] # Index within cluster
+        for k in range(n_clusters):
+            intra_cluster_medoid_index.append(
+                np.exp(-distances_k[k]/distances_k[k].std()).sum(axis=1).argmax()
+            )
+            # Here we need to use the global sample index to find the medoid structure:
+            medoid_index.append(cluster_indices_core[k][intra_cluster_medoid_index[k]])    
```

Comment:
I see this is pretty similar to the logic used below for the non-core option. The way it is written makes things more understandable since we have variables like `cluster_indices_core` which lets us know we're working with the core point indices, but the logic could be shorted by taking this code block out of the if/else statement. But not necessary!